### ISSUE TITLE
 Illegal instruction

### ISSUE THREAD

**Issue by Tonmoi123** [spyder-ide/spyder#20880]
### What steps will reproduce the problem?

Cant load a python file showing illegal hardware instruction

## Versions

* Spyder version: 5.4.3  (conda)
* Python version: 3.11.3 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.7
* Operating System: Darwin 21.5.0

### Dependencies

```
# Mandatory:
applaunchservices >=0.3.0                        :  0.3.0 (OK)
atomicwrites >=1.2.0                             :  1.4.0 (OK)
chardet >=2.0.0                                  :  4.0.0 (OK)
cloudpickle >=0.5.0                              :  2.0.0 (OK)
cookiecutter >=1.6.0                             :  1.7.3 (OK)
diff_match_patch >=20181111                      :  20200713 (OK)
intervaltree >=3.0.2                             :  3.1.0 (OK)
IPython >=7.31.1,<9.0.0,!=8.8.0,!=8.9.0,!=8.10.0 :  8.12.0 (OK)
jedi >=0.17.2,<0.19.0                            :  0.18.1 (OK)
jellyfish >=0.7                                  :  0.9.0 (OK)
jsonschema >=3.2.0                               :  4.17.3 (OK)
keyring >=17.0.0                                 :  23.13.1 (OK)
nbconvert >=4.0                                  :  6.5.4 (OK)
numpydoc >=0.6.0                                 :  1.5.0 (OK)
parso >=0.7.0,<0.9.0                             :  0.8.3 (OK)
pexpect >=4.4.0                                  :  4.8.0 (OK)
pickleshare >=0.4                                :  0.7.5 (OK)
psutil >=5.3                                     :  5.9.0 (OK)
pygments >=2.0                                   :  2.11.2 (OK)
pylint >=2.5.0,<3.0                              :  2.16.2 (OK)
pylint_venv >=2.1.1                              :  2.3.0 (OK)
pyls_spyder >=0.4.0                              :  0.4.0 (OK)
pylsp >=1.7.2,<1.8.0                             :  1.7.2 (OK)
pylsp_black >=1.2.0                              :  1.2.1 (OK)
qdarkstyle >=3.0.2,<3.2.0                        :  3.0.2 (OK)
qstylizer >=0.2.2                                :  0.2.2 (OK)
qtawesome >=1.2.1                                :  1.2.2 (OK)
qtconsole >=5.4.2,<5.5.0                         :  5.4.2 (OK)
qtpy >=2.1.0                                     :  2.2.0 (OK)
rtree >=0.9.7                                    :  1.0.1 (OK)
setuptools >=49.6.0                              :  66.0.0 (OK)
sphinx >=0.6.6                                   :  5.0.2 (OK)
spyder_kernels >=2.4.3,<2.5.0                    :  2.4.3 (OK)
textdistance >=4.2.0                             :  4.2.1 (OK)
three_merge >=0.1.1                              :  0.1.1 (OK)
watchdog >=0.10.3                                :  2.1.6 (OK)
zmq >=22.1.0                                     :  23.2.0 (OK)

# Optional:
cython >=0.21                                    :  None (NOK)
matplotlib >=3.0.0                               :  None (NOK)
numpy >=1.7                                      :  1.23.5 (OK)
pandas >=1.1.1                                   :  None (NOK)
scipy >=0.17.0                                   :  1.10.1 (OK)
sympy >=0.7.3                                    :  None (NOK)
```

### Environment

<details><summary>Environment</summary>

```
# packages in environment at /Users/tonmoibora/opt/anaconda3/envs/virtual_platform:
#
# Name                    Version                   Build  Channel
absl-py                   1.4.0                    pypi_0    pypi
alabaster                 0.7.12             pyhd3eb1b0_0  
applaunchservices         0.3.0           py311hecd8cb5_0  
appnope                   0.1.2           py311hecd8cb5_1001  
arrow                     1.2.3           py311hecd8cb5_1  
astroid                   2.14.2          py311hecd8cb5_0  
asttokens                 2.0.5              pyhd3eb1b0_0  
astunparse                1.6.3                    pypi_0    pypi
atomicwrites              1.4.0                      py_0  
attrs                     22.1.0          py311hecd8cb5_0  
autopep8                  1.6.0              pyhd3eb1b0_1  
babel                     2.11.0          py311hecd8cb5_0  
backcall                  0.2.0              pyhd3eb1b0_0  
beautifulsoup4            4.12.2          py311hecd8cb5_0  
binaryornot               0.4.4              pyhd3eb1b0_1  
black                     23.3.0          py311hecd8cb5_0  
blas                      1.0                         mkl  
bleach                    4.1.0              pyhd3eb1b0_0  
brotlipy                  0.7.0           py311h6c40b1e_1002  
bzip2                     1.0.8                h1de35cc_0  
ca-certificates           2023.01.10           hecd8cb5_0  
cachetools                5.3.0                    pypi_0    pypi
certifi                   2022.12.7       py311hecd8cb5_0  
cffi                      1.15.1          py311h6c40b1e_3  
chardet                   4.0.0           py311hecd8cb5_1003  
charset-normalizer        2.0.4              pyhd3eb1b0_0  
click                     8.0.4           py311hecd8cb5_0  
cloudpickle               2.0.0              pyhd3eb1b0_0  
colorama                  0.4.6           py311hecd8cb5_0  
comm                      0.1.2           py311hecd8cb5_0  
cookiecutter              1.7.3              pyhd3eb1b0_0  
cryptography              39.0.1          py311hf6deb26_0  
debugpy                   1.5.1           py311hcec6c5f_0  
decorator                 5.1.1              pyhd3eb1b0_0  
defusedxml                0.7.1              pyhd3eb1b0_0  
diff-match-patch          20200713           pyhd3eb1b0_0  
dill                      0.3.6           py311hecd8cb5_0  
docstring-to-markdown     0.11            py311hecd8cb5_0  
docutils                  0.18.1          py311hecd8cb5_3  
entrypoints               0.4             py311hecd8cb5_0  
executing                 0.8.3              pyhd3eb1b0_0  
flake8                    6.0.0           py311hecd8cb5_0  
flatbuffers               23.3.3                   pypi_0    pypi
freetype                  2.12.1               hd8bbffd_0  
gast                      0.4.0                    pypi_0    pypi
gettext                   0.21.0               he85b6c0_1  
giflib                    5.2.1                h6c40b1e_3  
glib                      2.69.1               hfff2838_2  
google-auth               2.17.3                   pypi_0    pypi
google-auth-oauthlib      1.0.0                    pypi_0    pypi
google-pasta              0.2.0                    pypi_0    pypi
grpcio                    1.54.0                   pypi_0    pypi
gst-plugins-base          1.14.1               hcec6c5f_1  
gstreamer                 1.14.1               h6c40b1e_1  
h5py                      3.8.0                    pypi_0    pypi
icu                       58.2                 h0a44026_3  
idna                      3.4             py311hecd8cb5_0  
imagesize                 1.4.1           py311hecd8cb5_0  
importlib-metadata        6.0.0           py311hecd8cb5_0  
importlib_metadata        6.0.0                hd3eb1b0_0  
inflection                0.5.1           py311hecd8cb5_0  
intel-openmp              2021.4.0          hecd8cb5_3538  
intervaltree              3.1.0              pyhd3eb1b0_0  
ipykernel                 6.19.2          py311h4975b63_0  
ipython                   8.12.0          py311hecd8cb5_0  
ipython_genutils          0.2.0              pyhd3eb1b0_1  
isort                     5.9.3              pyhd3eb1b0_0  
jaraco.classes            3.2.1              pyhd3eb1b0_0  
jax                       0.4.8                    pypi_0    pypi
jedi                      0.18.1          py311hecd8cb5_1  
jellyfish                 0.9.0           py311h6c40b1e_0  
jinja2                    3.1.2           py311hecd8cb5_0  
jinja2-time               0.2.0              pyhd3eb1b0_3  
jpeg                      9e                   h6c40b1e_1  
jsonschema                4.17.3          py311hecd8cb5_0  
jupyter_client            8.1.0           py311hecd8cb5_0  
jupyter_core              5.3.0           py311hecd8cb5_0  
jupyterlab_pygments       0.1.2                      py_0  
keras                     2.12.0                   pypi_0    pypi
keyring                   23.13.1         py311hecd8cb5_0  
krb5                      1.19.4               hdba6334_0  
lazy-object-proxy         1.6.0           py311h6c40b1e_0  
lcms2                     2.12                 hf1fd2bf_0  
lerc                      3.0                  he9d5cce_0  
libclang                  16.0.0                   pypi_0    pypi
libclang13                14.0.6          default_habbcc1a_1  
libcxx                    14.0.6               h9765a3e_0  
libdeflate                1.17                 hb664fd8_0  
libedit                   3.1.20221030         h6c40b1e_0  
libffi                    3.4.2                hecd8cb5_6  
libiconv                  1.16                 hca72f7f_2  
libllvm14                 14.0.6               h91fad77_2  
libpng                    1.6.39               h6c40b1e_0  
libpq                     12.9                 h1c9f633_3  
libsodium                 1.0.18               h1de35cc_0  
libspatialindex           1.9.3                h23ab428_0  
libtiff                   4.5.0                hcec6c5f_2  
libwebp                   1.2.4                hf6ce154_1  
libwebp-base              1.2.4                h6c40b1e_1  
libxml2                   2.10.3               h930c0e2_0  
libxslt                   1.1.37               h6d1eb0e_0  
llvm-openmp               14.0.6               h0dcd299_0  
lxml                      4.9.2           py311h6c40b1e_0  
lz4-c                     1.9.4                hcec6c5f_0  
markdown                  3.4.3                    pypi_0    pypi
markupsafe                2.1.1           py311h6c40b1e_0  
matplotlib-inline         0.1.6           py311hecd8cb5_0  
mccabe                    0.7.0              pyhd3eb1b0_0  
mistune                   0.8.4           py311h6c40b1e_1000  
mkl                       2021.4.0           hecd8cb5_637  
mkl-service               2.4.0           py311h6c40b1e_0  
mkl_fft                   1.3.1           py311hbc8bb1e_0  
mkl_random                1.2.2           py311hc5848a5_0  
ml-dtypes                 0.1.0                    pypi_0    pypi
more-itertools            8.12.0             pyhd3eb1b0_0  
mypy_extensions           0.4.3           py311hecd8cb5_1  
nbclient                  0.5.13          py311hecd8cb5_0  
nbconvert                 6.5.4           py311hecd8cb5_0  
nbformat                  5.7.0           py311hecd8cb5_0  
ncurses                   6.4                  hcec6c5f_0  
nest-asyncio              1.5.6           py311hecd8cb5_0  
nspr                      4.33                 he9d5cce_0  
nss                       3.74                 h47edf6a_0  
numpy                     1.23.5                   pypi_0    pypi
numpydoc                  1.5.0           py311hecd8cb5_0  
oauthlib                  3.2.2                    pypi_0    pypi
openssl                   1.1.1t               hca72f7f_0  
opt-einsum                3.3.0                    pypi_0    pypi
packaging                 23.0            py311hecd8cb5_0  
pandocfilters             1.5.0              pyhd3eb1b0_0  
parso                     0.8.3              pyhd3eb1b0_0  
pathspec                  0.10.3          py311hecd8cb5_0  
pcre                      8.45                 h23ab428_0  
pexpect                   4.8.0              pyhd3eb1b0_3  
pickleshare               0.7.5           pyhd3eb1b0_1003  
pillow                    9.4.0           py311hcec6c5f_0  
pip                       23.0.1          py311hecd8cb5_0  
platformdirs              2.5.2           py311hecd8cb5_0  
pluggy                    1.0.0           py311hecd8cb5_1  
ply                       3.11            py311hecd8cb5_0  
poyo                      0.5.0              pyhd3eb1b0_0  
prompt-toolkit            3.0.36          py311hecd8cb5_0  
protobuf                  4.22.3                   pypi_0    pypi
psutil                    5.9.0           py311h6c40b1e_0  
ptyprocess                0.7.0              pyhd3eb1b0_2  
pure_eval                 0.2.2              pyhd3eb1b0_0  
pyasn1                    0.5.0                    pypi_0    pypi
pyasn1-modules            0.3.0                    pypi_0    pypi
pycodestyle               2.10.0          py311hecd8cb5_0  
pycparser                 2.21               pyhd3eb1b0_0  
pydocstyle                6.3.0           py311hecd8cb5_0  
pyflakes                  3.0.1           py311hecd8cb5_0  
pygments                  2.11.2             pyhd3eb1b0_0  
pylint                    2.16.2          py311hecd8cb5_0  
pylint-venv               2.3.0           py311hecd8cb5_0  
pyls-spyder               0.4.0              pyhd3eb1b0_0  
pyobjc-core               9.0             py311h9205ec4_1  
pyobjc-framework-cocoa    9.0             py311h9205ec4_0  
pyobjc-framework-coreservices 9.0             py311h46256e1_0  
pyobjc-framework-fsevents 9.0             py311hecd8cb5_0  
pyopenssl                 23.0.0          py311hecd8cb5_0  
pyqt                      5.15.7          py311hcec6c5f_0  
pyqt5-sip                 12.11.0         py311hcec6c5f_0  
pyqtwebengine             5.15.7          py311hcec6c5f_0  
pyrsistent                0.18.0          py311h6c40b1e_0  
pysocks                   1.7.1           py311hecd8cb5_0  
python                    3.11.3               h1fd4e5f_0  
python-dateutil           2.8.2              pyhd3eb1b0_0  
python-fastjsonschema     2.16.2          py311hecd8cb5_0  
python-lsp-black          1.2.1           py311hecd8cb5_0  
python-lsp-jsonrpc        1.0.0              pyhd3eb1b0_0  
python-lsp-server         1.7.2           py311hecd8cb5_0  
python-slugify            5.0.2              pyhd3eb1b0_0  
python.app                3               py311h6c40b1e_0  
pytoolconfig              1.2.5           py311hecd8cb5_1  
pytz                      2022.7          py311hecd8cb5_0  
pyyaml                    6.0             py311h6c40b1e_1  
pyzmq                     23.2.0          py311hcec6c5f_0  
qdarkstyle                3.0.2              pyhd3eb1b0_0  
qstylizer                 0.2.2           py311hecd8cb5_0  
qt-main                   5.15.2               h51e0635_8  
qt-webengine              5.15.9               h90a370e_4  
qtawesome                 1.2.2           py311hecd8cb5_0  
qtconsole                 5.4.2           py311hecd8cb5_0  
qtpy                      2.2.0           py311hecd8cb5_0  
qtwebkit                  5.212                hbfab81c_5  
readline                  8.2                  hca72f7f_0  
requests                  2.29.0          py311hecd8cb5_0  
requests-oauthlib         1.3.1                    pypi_0    pypi
rope                      1.7.0           py311hecd8cb5_0  
rsa                       4.9                      pypi_0    pypi
rtree                     1.0.1           py311hecd8cb5_0  
scipy                     1.10.1                   pypi_0    pypi
setuptools                66.0.0          py311hecd8cb5_0  
sip                       6.6.2           py311hcec6c5f_0  
six                       1.16.0             pyhd3eb1b0_1  
snowballstemmer           2.2.0              pyhd3eb1b0_0  
sortedcontainers          2.4.0              pyhd3eb1b0_0  
soupsieve                 2.4             py311hecd8cb5_0  
sphinx                    5.0.2           py311hecd8cb5_0  
sphinxcontrib-applehelp   1.0.2              pyhd3eb1b0_0  
sphinxcontrib-devhelp     1.0.2              pyhd3eb1b0_0  
sphinxcontrib-htmlhelp    2.0.0              pyhd3eb1b0_0  
sphinxcontrib-jsmath      1.0.1              pyhd3eb1b0_0  
sphinxcontrib-qthelp      1.0.3              pyhd3eb1b0_0  
sphinxcontrib-serializinghtml 1.1.5              pyhd3eb1b0_0  
spyder                    5.4.3           py311hecd8cb5_1  
spyder-kernels            2.4.3           py311hecd8cb5_0  
sqlite                    3.41.2               h6c40b1e_0  
stack_data                0.2.0              pyhd3eb1b0_0  
tensorboard               2.12.2                   pypi_0    pypi
tensorboard-data-server   0.7.0                    pypi_0    pypi
tensorboard-plugin-wit    1.8.1                    pypi_0    pypi
tensorflow                2.12.0                   pypi_0    pypi
tensorflow-estimator      2.12.0                   pypi_0    pypi
tensorflow-io-gcs-filesystem 0.32.0                   pypi_0    pypi
termcolor                 2.3.0                    pypi_0    pypi
text-unidecode            1.3                pyhd3eb1b0_0  
textdistance              4.2.1              pyhd3eb1b0_0  
three-merge               0.1.1              pyhd3eb1b0_0  
tinycss2                  1.2.1           py311hecd8cb5_0  
tk                        8.6.12               h5d9f67b_0  
toml                      0.10.2             pyhd3eb1b0_0  
tomlkit                   0.11.1          py311hecd8cb5_0  
tornado                   6.2             py311h6c40b1e_0  
traitlets                 5.7.1           py311hecd8cb5_0  
typing-extensions         4.5.0                    pypi_0    pypi
tzdata                    2023c                h04d1e81_0  
ujson                     5.4.0           py311hcec6c5f_0  
unidecode                 1.2.0              pyhd3eb1b0_0  
urllib3                   1.26.15         py311hecd8cb5_0  
watchdog                  2.1.6           py311h3333b6a_0  
wcwidth                   0.2.5              pyhd3eb1b0_0  
webencodings              0.5.1           py311hecd8cb5_1  
werkzeug                  2.3.1                    pypi_0    pypi
whatthepatch              1.0.2           py311hecd8cb5_0  
wheel                     0.38.4          py311hecd8cb5_0  
wrapt                     1.14.1          py311h6c40b1e_0  
wurlitzer                 3.0.2           py311hecd8cb5_0  
xz                        5.2.10               h6c40b1e_1  
yaml                      0.2.5                haf1e3a3_0  
yapf                      0.31.0             pyhd3eb1b0_0  
zeromq                    4.3.4                h23ab428_0  
zipp                      3.11.0          py311hecd8cb5_0  
zlib                      1.2.13               h4dc903c_0  
zstd                      1.5.5                hc035e20_0  

```
</details>

**Comment by ccordoba12** [comment]
Hey @Tonmoi123, thanks for reporting. Could you provide more details about this?

For instance, where are you trying to load the file? In the editor or the IPython console? Also, could you provide a simple example of the kind of file you're trying to load?

Thanks for your help!

**Comment by Tonmoi123** [comment]
Trying to run a python program but everytime I try to run this line showsIllegal hardware instruction Sent from my iPhoneOn 29-Apr-2023, at 8:21 PM, Carlos Cordoba ***@***.***> wrote:﻿
Hey @Tonmoi123, thanks for reporting. Could you provide more details about this?
For instance, where are you trying to load the file? In the editor or the IPython console? Also, could you provide a simple example of the kind of file you're trying to load?
Thanks for your help!

**Comment by ccordoba12** [comment]
I see. Then I think this is probably a duplicate of issue #20800, which will be solved in our next version (5.4.4), to be released in June.

Could you confirm that?

**Comment by ccordoba12** [comment]
Closing due to lack of response.
